src/utils/parseGlobalAccessToken.ts

```diff
@@ -0,0 +1,27 @@
+import { get } from "lodash";
+import type { Settings } from "../types";
+
+type ParseGlobalAccessToken = (tokens?: Settings["global_access_token"]) => {
+  access_token: string;
+  refresh_token: string;
+};
+
+const parseGlobalAccessToken: ParseGlobalAccessToken = (tokens) => {
+  const parsedTokens = { access_token: "", refresh_token: "" };
+
+  if (!tokens) {
+    return parsedTokens;
+  }
+
+  try {
+    const parsed = JSON.parse(tokens);
+    parsedTokens.access_token = get(parsed, "access_token", "");
+    parsedTokens.refresh_token = get(parsed, "refresh_token", "");
+  } catch (e) {
+    //..
+  }
+
+  return parsedTokens;
+};
+
+export { parseGlobalAccessToken };
```

Comment:
If invalid tokens are passed in, then it returns an object with `refresh` and `access` tokens which makes it look likes it works. Instead, I think it should throw an exception so it's clear it failed. The calling code can then catch that do what it wants (even if it is nothing/ignores it).

Reply:
It's almost impossible for the wrong JSON to come here. And even if so, then the API will provide a clear error message

Reply:
I'm confused. If it never fails, then why add any checking and if it does fail, why not throw and handle the exception? If there was an error, the API may throw a clear error message but even if it did, I don't see anywhere where that's logged/saved/shown - unless you opened your network tab in the browser before making the requests.

Reply:
For better UX. I don't want to burden the user with unnecessary error messages.

Reply:
I'm not sure I understand. The user did something to get some authentication tokens, it then failed (even though it is unlikely as you mentioned) and so we did nothing.
The user is surely going to be expecting to see some feedback like "Failed to authenticate" but instead we're silently ignoring the issue and hoping everything contains happily?

My concern is the user thinks everything is working fine, but in reality it's either working unexpectedly or not at all...